### ISSUE TITLE
Add Japanese to the language switcher

### ISSUE THREAD

**Issue by JulienPalard** [python/cpython#75332]
BPO | [31149](https://bugs.python.org/issue31149)
--- | :---
Nosy | @vstinner, @ned-deily, @methane, @JulienPalard
PRs | <li>python/cpython#3028</li><li>python/cpython#3052</li><li>python/cpython#3081</li><li>python/cpython#3114</li><li>python/cpython#3115</li><li>python/cpython#3117</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-22.11:28:56.075>
created_at = <Date 2017-08-08.20:58:44.692>
labels = ['3.7']
title = 'Add Japanese to the language switcher'
updated_at = <Date 2017-09-22.11:28:56.074>
user = 'https://github.com/JulienPalard'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-22.11:28:56.074>
actor = 'mdk'
assignee = 'none'
closed = True
closed_date = <Date 2017-09-22.11:28:56.075>
closer = 'mdk'
components = []
creation = <Date 2017-08-08.20:58:44.692>
creator = 'mdk'
dependencies = []
files = []
hgrepos = []
issue_num = 31149
keywords = []
message_count = 10.0
messages = ['299947', '299949', '300094', '300096', '300213', '300216', '300231', '300242', '300265', '300434']
nosy_count = 5.0
nosy_names = ['vstinner', 'ned.deily', 'Arfrever', 'methane', 'mdk']
pr_nums = ['3028', '3052', '3081', '3114', '3115', '3117']
priority = 'normal'
resolution = None
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31149'
versions = ['Python 2.7', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by JulienPalard** [comment]
Japanese translation have reached the prerequisites from PEP-545, their last step is to be added to the language switcher.

Current build has been reviewed by Inada and is OK: https://docs.python.org/ja/3.7/

Note: On the current build, the /ja/ is not detected by the language switcher, as this \*is* the step to add /ja/ to this same language switcher, this is the issue http://bugs.python.org/issue31146

**Comment by vstinner** [comment]
New changeset c82b7f332aff606af6c9c163da75f1e86514125e by Victor Stinner (Julien Palard) in branch 'master':
bpo-31149: Doc: Add Japanese to the language switcher. (bpo-3028)
https://github.com/python/cpython/commit/c82b7f332aff606af6c9c163da75f1e86514125e

**Comment by vstinner** [comment]
New changeset e8e7fba0b24582959feca9c31f2a72fc0251f83d by Victor Stinner (Julien Palard) in branch '3.6':
bpo-31159: fix language switch regex on unknown yet built languages. … (bpo-3051)
https://github.com/python/cpython/commit/e8e7fba0b24582959feca9c31f2a72fc0251f83d

**Comment by vstinner** [comment]
I reopen the issue: backport to 2.7 is still needed.

**Comment by 80036ac5-bb84-4d39-8416-02cd8e51707d** [comment]
There is a minor inconsistency:
For French language, name in the list is displayed in French as "Français".
For Japanese language, name in the list is displayed in English as "Japanese" instead of in Japanese as "日本語".

Maybe it would be best to display 2 names for each non-English language? Example:
 "Français (French)"
 "日本語 (Japanese)"

**Comment by JulienPalard** [comment]
Arfrever: nice catch!

Inda: What do you think? Any preference between "日本語", "Japanese", and  "日本語 (Japanese)"?

I personally don't like much the "日本語 (Japanese)" version, but I don't care much between "French" and "Français" as they're using the same alphabet.

Maybe displaying "日本語" instead of "Japanese" show more commitment into translations? Like displaying "Japanese" meaning "OK we translated, but not everything, starting by your language".

Another solution would be to translate each languages in each languages, and display "Anglais, Japonais, Français" in the french translation, "English, Japanese, French" in the english one, and so on…

I think I prefer to display languages in their own language, so "English, "Français", "日本語" for me.

**Comment by vstinner** [comment]
New changeset fe8d9dc479a96ef490034107e7d4a6228b4be140 by Victor Stinner (Julien Palard) in branch '2.7':
bpo-31159: fix language switch regex on unknown yet built languages. … (bpo-3051) (bpo-3081)
https://github.com/python/cpython/commit/fe8d9dc479a96ef490034107e7d4a6228b4be140

**Comment by methane** [comment]
My preference is "Japanese".

After selecting drop down list, we can type "jap" to select Japanese.
On the other hand, we should use IME to input "日本語".  It's not so easy as input alphabets.

**Comment by JulienPalard** [comment]
@inada make sense, so I'll change "Français" for "French" for consistency.

**Comment by JulienPalard** [comment]
Has been merged on master, backported to 3.6 and 2.7.

It starts to appear on some versions as the script is currently running, like in https://docs.python.org/2.7/.

I'll close when I'll see this on all versions.
